File: Ch05/05_04/urlcheck.py
from urllib.request import URLError, urlopen


def check_url(url):
    try:
        with urlopen(url, timeout=5):
            return True
    except (URLError, TimeoutError):
        return False


if __name__ == '__main__':
    url = 'https://www.linkedin.com/learning/'
    print(check_url(url))

File: Ch05/05_04/urlcheck.go
package main

import (
	"context"
	"net/http"
	"time"
)

func CheckURL(url string) bool {
	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
	defer cancel()

	req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
	if err != nil {
		return false
	}

	resp, err := http.DefaultClient.Do(req)
	if err != nil {
		return false
	}

	if resp.StatusCode != http.StatusOK {
		return false
	}

	return true
}

func main() {
	url := "https://www.linkedin.com/learning/"
	print(CheckURL(url))
}
